R/test_compliance.R

```diff
@@ -25,14 +25,21 @@ test_compliance <- function(skip = NULL, ctx = get_default_context()) {
 
       where <- asNamespace(pkg)
 
-      classes <- sort(getClasses(where))
-      expect_equal(length(classes), length(key_methods))
+      sapply(names(key_methods), function(name) {
+        dbi_class <- paste0("DBI", name)
+        
+        classes <- Filter(function(class) {
+          extends(class, dbi_class) && getClass(class)@virtual == FALSE
+        }, getClasses(where))
```

Comment:
Do you need `#' @importFrom methods getClass getClasses`? Then it might work even without `library(methods)`, which I'd prefer.